### PR TITLE
feat(helm): add support for NetworkPolicies in v2 Helm chart

### LINKED ISSUE #3160: Add support for NetworkPolicy to the v2 chart
**Describe the current behavior**
When deploying the Azure Service Operator on a cluster with default deny NetworkPolicies for every namespace then the Kubernetes API service is unable to communicate with the operator pod.

**Describe the improvement**
Add support for deploying NetworkPolicies, which allow Ingress traffic to the pod and Egress from the pod.

**Additional context**
I have the required NetworkPolicies ready, but I'm unsure how the release process of the Helm chart works right now and would need some guidance on how to contribute this change.

### PR DESCRIPTION
**Pull request by tongpu**
Closes #3160

**What this PR does / why we need it**:

Adds support for NetworkPolicies to ASO v2 Helm chart

**Special notes for your reviewer**:

**How does this PR make you feel**:
gif

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

**Comment by tongpu** [comment]
@microsoft-github-policy-service agree company="Adfinis"

File: v2/charts/azure-service-operator/templates/networkpolicies.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: azure-service-operator-allow-egress
+  namespace: {{ .Release.namespace }}
+spec:
+  egress:
+  - ports:
+    # Required for communication with the Azure API
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/charts/azure-service-operator/templates/networkpolicies.yaml:26]: As with the Kubernetes APIServer, it'd be good if they were parameterized in values.

**Review by matthchr — COMMENTED** [review]
Small comment, but other than that looks good to me.

We need to add another label to the pod as well, but we'll do that out of band. I'll file a separate issue for it.

File: v2/charts/azure-service-operator/templates/networkpolicies.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tongpu) [reply v2/charts/azure-service-operator/templates/networkpolicies.yaml:26]: > We need to add another label to the pod as well, but we'll do that out of band. I'll file a separate issue for it.

I wanted to propose that as well. Adding the usual `app.kubernetes.io` labels would be in line with what other Helm charts do.

What I was also thinking about is to make the destination ip ranges configurable per port, allowing people to override the default `0.0.0.0/0`, which is rather open. ;)

**Comment by matthchr** [comment]
@tongpu is this still something you're driving? If not we can probably take it over and get it merged. Let us know.

**Comment by tongpu** [comment]
> @tongpu is this still something you're driving? If not we can probably take it over and get it merged. Let us know.

I've lost track of it as life was rather busy recently. I'll push the proposed changes by next week the latest.

**Comment by tongpu** [comment]
@matthchr Thanks for the reminder. I've pushed the remaining changes and I think the PR should be ready now.

File: v2/charts/azure-service-operator/values.yaml [hunk 2]
```diff
@@ -120,3 +120,15 @@ podAnnotations: {}
...
+
+# networkPolicies allows you to configure the NetworkPolicies deployed as part of the Chart
+networkPolicies:
+  enable: true
+  # TCP port to be configured for talking to the Kubernetes API
+  kubernetesApiPort: 6443
+  # Destination CIDR for talking to the Kubernetes API
+  kubernetesApiCIDR: 0.0.0.0/0
+  # Destination CIDR for talking to MySQL servers
+  kubernetesApiCIDR: 0.0.0.0/0
```
Comment (matthchr) [inline v2/charts/azure-service-operator/values.yaml:132]: copy/paste error (caught by CI): This should be `mysqlCIDR`, and the one below `postgresqlCIDR`

Comment (tongpu) [reply v2/charts/azure-service-operator/values.yaml]: Fixed. One should not commit on a Friday afternoon.

File: v2/charts/azure-service-operator/templates/networkpolicies.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      protocol: TCP
+    to:
+    - ipBlock:
+        cidr: 0.0.0.0/0
+    # Required for communication with the Kubernetes API
+    - port: {{ .Values.networkPolicies.kubernetesApiPort }}
+      protocol: TCP
+    to:
+    - ipBlock:
+        cidr: {{ .Values.networkpolicies.kubernetesApiCIDR }}
```
Comment (matthchr) [inline v2/charts/azure-service-operator/templates/networkpolicies.yaml:37]: Ahh, I missed CI was also having issue with `networkpolicies` vs `networkPolicies`.

It's supposed to be `networkPolicies` but a few of these here have a lowercase `p`

**Comment by matthchr** [comment]
live mysql test timed out.. retrying

**Review by matthchr — APPROVED** [review]
Doh, forgot to approve earlier.

**Comment by matthchr** [comment]
Thanks for the contribution @tongpu!

**Pull request merged**